src/qt/addresstablemodel.cpp

```diff
@@ -196,7 +196,7 @@ QVariant AddressTableModel::data(const QModelIndex &index, int role) const
     if(!index.isValid())
         return QVariant();
 
-    AddressTableEntry *rec = static_cast<AddressTableEntry*>(index.internalPointer());
+    const AddressTableEntry* rec{priv->index(index.row())};
```

Comment:
Need to check if `rec != nullptr`, `index.isValid()` doesn't check for out of bounds.

Alternatively you could inline `priv->index` and use reference instead `const AddressTableEntry&`. You still need to check for out of bounds.